### PR TITLE
Use href="javascript:;" to indicate anchor tags without link #7592

### LINKED ISSUE #7592: Use href="javascript:;" to indicate anchor tags without link
```html
<a href="#">Text</a>
```
The above is commonly used such that `Text` gains the appearance of a link while it does not actually link anywhere. However, this is the wrong method because `#` actually links to the top of the web page. The correct way to suppress such link is with `href="javascript:;"`.

P.S. Of course, there might be cases where the intention _is_ to go to the top of the page. This needs to be examined case-by-case.

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #7592 

**Outline of Solution**

Replaced `href="#"` with `href="javascript:;"`

### CODE REVIEW

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/partials/instructorHelpAddEditInstructors.jsp [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
                                 <a href="javascript:;" id="instrEditLink3" class="btn btn-primary btn-xs" data-toggle="tooltip" data-placement="top" title="Edit instructor details" disabled="">
                                     <span class="glyphicon glyphicon-pencil"></span> Edit</a>
-                                <a href="#" id="instrDeleteLink3" class="btn btn-primary btn-xs" data-toggle="tooltip" data-placement="top" title="Delete the instructor from the course" disabled="">
+                                <a href="" id="instrDeleteLink3" class="btn btn-primary btn-xs" data-toggle="tooltip" data-placement="top" title="Delete the instructor from the course" disabled="">
```
Comment (wkurniawan07) [inline src/main/webapp/partials/instructorHelpAddEditInstructors.jsp]: Looks like this one is not correctly replaced?

**Comment by whipermr5** [comment]
@amarlearning Pending build errors

File: src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag [hunk 2]
```diff
@@ -50,7 +50,7 @@
      data-toggle="tooltip"
      data-placement="top"
      style="display: inline-block; padding-right: 5px;">
-     <div class="btn-group margin-bottom-7px">
+    <div class="btn-group margin-bottom-7px">
```
Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: This is an unrelated change introduced when fixing the merge conflict.

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: EDIT: I'm okay with leaving it as it's actually correcting the indentation. The wrong indentation was introduced recently in #7312.

Comment (amarlearning) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: Will fix this.

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: You mean that the wrong indentation was not introduced here, but in #7312 right? I'm okay with allowing this change here so that the indentation can be corrected.

File: src/test/resources/pages/instructorFeedbackDeleteSuccessful.html [hunk 3]
```diff
@@ -1014,7 +1014,7 @@ <h5 class="col-xs-2" data-original-title="Select this option to use a custom tim
                     </a>
                   </li>
                   <li>
-                    <a class="session-publish-for-test" data-fsname="private session of characters1234567 #" data-sending-published-email="true" href="/page/instructorFeedbackPublish?courseid=CFeedbackUiT.CS2104&fsname=private+session+of+characters1234567+%23&next=%2Fpage%2FinstructorFeedbackSessionsPage&user=CFeedbackUiT.instructor&token=${sessionToken}">
+                    <a class="session-publish-for-test" data-fsname="private session of characters1234567 #" data-sending-published-email="true" disabled="" href="/page/instructorFeedbackPublish?courseid=CFeedbackUiT.CS2104&fsname=private+session+of+characters1234567+%23&next=%2Fpage%2FinstructorFeedbackSessionsPage&user=CFeedbackUiT.instructor&token=${sessionToken}">
```
Comment (whipermr5) [inline src/test/resources/pages/instructorFeedbackDeleteSuccessful.html]: Revert this; it's causing the build failure. This test has a dependency on the browser time zone; it'll fail when it (significantly?) differs from UTC.

File: src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html [hunk 4]
```diff
@@ -216,4 +216,4 @@
       </div>
     </div>
   </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: Unrelated change.

File: src/test/resources/pages/instructorFeedbackDeleteSuccessful.html (hunk 3 above)
Comment (amarlearning) [inline src/test/resources/pages/instructorFeedbackDeleteSuccessful.html]: @whipermr5 how to fix this?

Comment (whipermr5) [inline src/test/resources/pages/instructorFeedbackDeleteSuccessful.html]: It's a separate issue. Currently the only way to pass the test locally is to set your OS time to UTC. As long as it's passing on Travis, you're fine.

File: src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag (hunk 2 above)
Comment (amarlearning) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: @whipermr5 I don't see any intendation problem here :thinking:

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: Your current changeset is correct. The one in master is [currently wrong](https://github.com/TEAMMATES/teammates/blob/d0b4dd17f22eb18d1f325201a440b32045aa6859/src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag#L53) (5 spaces to the left of `div` instead of 4) due to #7312.

Comment (whipermr5) [inline src/main/webapp/WEB-INF/tags/instructor/feedbacks/feedbackSessionActions.tag]: Therefore I'm allowing this unrelated change to fix the one in master.

File: src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html (hunk 4 above)
Comment (amarlearning) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: can't see any change here :sweat_smile:

Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: I think the one in master doesn't end with a newline character.

Comment (wkurniawan07) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: @whipermr5 the correct God mode behaviour should be _with_ a newline character in the end. I'm guessing someone did not run God mode properly (or did not use God mode) before.

Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: I see. Maybe we should include test HTML files in the lintspaces config too?

Comment (amarlearning) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: @wkurniawan07 is right! maybe it's because of #7312. I am responsible for that. But now it's fixed.
sorry about that!

Comment (wkurniawan07) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: @whipermr5 there's a reason I did not; you can give it a try yourself.
(of course, you can have another lintspaces config for HTML test files)

Comment (whipermr5) [inline src/test/resources/pages/instructorSearchPageSearchFeedbackResponseComments.html]: I'm guessing they have more than one newline between blocks and/or trailing spaces.
